code/datums/components/pellet_cloud.dm

```diff
@@ -244,9 +245,15 @@
 				wound_info_by_part[hit_part][CLOUD_POSITION_W_BONUS] += P.wound_bonus
 				wound_info_by_part[hit_part][CLOUD_POSITION_BW_BONUS] += P.bare_wound_bonus
 				P.wound_bonus = CANT_WOUND // actual wounding will be handled aggregate
-
-	targets_hit[target]++
-	if(targets_hit[target] == 1)
+	else if(isobj(target))
+		var/obj/hit_object = target
+		playsound(hit_object, P.hitsound, 50, TRUE)
+		if(!hit_object.take_damage(P.damage, P.damage_type, P.flag, 0, turn(P.dir, 180), P.armour_penetration))
+			no_damage = TRUE
```

Comment:
You seem to be mistaken about pellet clouds not being able to damage objects? I just tested a shotgun loaded with buckshot on a normal window at pointblank range and did the expected ~42 damage (6 pellets * ~7 damage) to it. 
[![dreamseeker_2021-05-08_01-48-27.png](https://i.imgur.com/SGIKGZ2l.jpg)](https://i.imgur.com/SGIKGZ2.png)

Each pellet still handles its own damage dealing (and impact sound) in bullet_act(), this is just duplicating those so the pellets will deal damage and play their impact sounds twice. If you thought they weren't dealing damage because you're trying to use buckshot to open an airlock or reinforced window, that's because the individual pellets don't do enough damage to clear the damage_deflection, which wouldn't be affected by making them take damage twice.

Reply:
I'll test this again but I was pretty thorough about testing it and I'm pretty sure when I did pellet clouds weren't doing anything to regular windows. I'm aware of the damage_deflection on airlocks and reinforced windows.

Reply:
You're right, I fucked this up. They already do work on objects.